### PR TITLE
feat: Added SECRET_KEY as an environment export.

### LINKED ISSUE #84: Get SECRET_KEY as an export environment variable
## Description
As an owner of the project,
I need `SECRET_KEY` to not be sourced controlled,
so that I can configure it per environment without exposing to the Open Source community.

The `SECRET_KEY` env variable has to be exported instead of being hardcoded on the project.

### Update [Required]
- [ ] Get `SECRET_KEY` from machine environment
- [ ] Document this required environment variable on README
- [ ] Raise value error if this var is not set
- [ ] Write at least 1 unit test for this

## Definition of Done
- [ ] All of the required items are completed.
- [x] Approval by 1 mentor.

## Estimation
1 hour

### PR DESCRIPTION
**Pull request by sahil9001**
### Description

Seperated `SECRET_KEY` from the hardcoded data in the `settings.py` and added unit test for the same.

Fixes #84 

### Type of Change:

- Code
- Documentation

**Code/Quality Assurance Only**
- This change requires a documentation update (software upgrade on readme file)
- New feature (non-breaking change which adds functionality pre-approved by mentors)

### How Has This Been Tested?
Ran and all tests passed locally.
```
Creating test database for alias 'default'...
System check identified no issues (0 silenced).
.....
----------------------------------------------------------------------
Ran 5 tests in 0.158s


OK
Destroying test database for alias 'default'...
```

### Checklist:

- [x] My PR follows the style guidelines of this project
- [x] I have performed a self-review of my own code or materials
- [x] I have commented my code or provided relevant documentation, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] Any dependent changes have been merged
- [ ] I have written Kotlin Docs whenever is applicable

**Code/Quality Assurance Only**
- [x] My changes generate no new warnings
- [ ] My PR currently breaks something (fix or feature that would cause existing functionality to not work as expected)
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been published in downstream modules

### CODE REVIEW

**Comment by sahil9001** [comment]
@isabelcosta @codesankalp

File: tests/test_app_config.py
Comment (codesankalp) [inline tests/test_app_config.py]: this is not needed.
Don't import `unittest` and remove these last lines also because the test will run without this.

File: tests/test_app_config.py
Comment (codesankalp) [inline tests/test_app_config.py]:
```suggestion
from django.conf import settings
```
Use this.

**Review by codesankalp — DISMISSED** [review]
**Code Review**: Done

**Testing**:

**Test 1**: Without exporting `SECRET_KEY`:
       Screenshot:
       image
       Expected result: Value Error.
       Actual result: Value Error.

**Test 2**: Running server with exporting `SECRET_KEY`
Screenshot: 
image
Expected Result: Server will run.
Actual Result: Server runs perfectly after exporting it.

**Test 3**: Testing the tests
Screenshot:
image
Expected Result: tests run successfully without any error.
Actual Result: tests ran successfully without any error.

Thanks, @sahil9001 for your contribution.
LGTM :tada:

**Comment by codesankalp** [comment]
@isabelcosta needs one more review before merging. Can you review it?

File: tests/test_app_config.py
Comment (isabelcosta) [inline tests/test_app_config.py]: I don't quite understand this test, is this assuming you have SECRET_KEY as an environment variable? The tests should not depend on environment variables. Will this fail if I don't set up in the GitHub Actions?

Comment (sahil9001) [reply tests/test_app_config.py]: Umm.. I don't know the other way, can you please elaborate? I guess setting up unit tests for the `SECRET_KEY` variable was the thing to do, is there any other way to do this?

Comment (codesankalp) [reply tests/test_app_config.py]: > I don't quite understand this test, is this assuming you have SECRET_KEY as an environment variable? The tests should not depend on environment variables. Will this fail if I don't set up in the GitHub Actions?


Yes, it will fail in GitHub actions if `SECRET_KEY` is not exported as an environment variable.
I also don't recommend this test because there is no need of testing this feature @isabelcosta but the issue description consists of writing a test.

Comment (isabelcosta) [reply tests/test_app_config.py]: On mentorship backend we have a test for this, wherein setup of the test we add the variable. but i honestly do not know if this is a good practice or not, for testing environment variables: https://github.com/anitab-org/mentorship-backend/blob/develop/tests/test_app_config.py

I think you could add a setup the secret key on the `os.environ` and another test where you don't do that, and you make sure that the value error gets raised 🤔 

If you think there is no way around this, feel free to remove the test, I am actually not sure here

Comment (sahil9001) [reply tests/test_app_config.py]: Okay @isabelcosta ! I'll try that.

Comment (codesankalp) [reply tests/test_app_config.py]: @sahil9001 Are you facing any problems with this?

Comment (sahil9001) [reply tests/test_app_config.py]: Hey! I'm sorry I got busy with exams, will do it by today!

**Comment by sahil9001** [comment]
To be clear, I didn't dismiss any reviews, I just pushed my commit via github CLI. @codesankalp @isabelcosta

**Comment by Rahulm2310** [comment]
>To be clear, I didn't dismiss any reviews, I just pushed my commit via github CLI. @codesankalp @isabelcosta

@sahil9001 don't worry, it is the default github behavior.

File: tests/test_app_config.py
Comment (codesankalp) [reply tests/test_app_config.py]: @sahil9001 Is there any problem in implementing @isabelcosta Idea.
I think we need a test for settings.py and if we will add new environment variables in the future then this test can be extended to use that.

Comment (sahil9001) [reply tests/test_app_config.py]: @isabelcosta said to remove it, so I did, should I write a test case?

File: .github/workflows/tests.yml
Comment (codesankalp) [inline .github/workflows/tests.yml]: There is no secret key defined.
Use Github Workflow secret_key in this.
and why are you appending it to the test?

File: README.md
Comment (codesankalp) [inline README.md]: Actually, If we use dotenv then there is no need of using the export keyword in the file. We can directly define the environment variables.
Or If we don't want to add it in the .env file then we can simply export it.
I hope it makes clear to you.
So Please change the third point and make it more descriptive.

**Review by codesankalp — CHANGES_REQUESTED** [review]

File: .github/workflows/tests.yml
Comment (sahil9001) [reply .github/workflows/tests.yml]: I'm appending "test" as a dummy key to make sure that the Github actions pass. I'll add Github workflow `secrets.SECRET_KEY`.

**Comment by sahil9001** [comment]
Hey @codesankalp I'm having trouble in writing the proper command in `tests.yml` can you please guide me on that?

File: .github/workflows/tests.yml
Comment (codesankalp) [inline .github/workflows/tests.yml]: Don't use env here

```
- name: Exporting Secret Key
  run: export SECRET_KEY=${{ secrets.GITHUB_TOKEN }}
```

Comment (codesankalp) [reply .github/workflows/tests.yml]: And remove the other lines.

File: README.md
Comment (codesankalp) [inline README.md]:
```suggestion
3. `SECRET_KEY`  - This environment variable is required for running the backend. Add `SECRET_KEY` in `.env` file or export it by using `export SECRET_KEY=<YOUR SECRET KEY>`.
```
You can make it more descriptive.

**Review by codesankalp — CHANGES_REQUESTED** [review]
@sahil9001 I have suggested some changes which will solve your problem of running tests.

**Comment by sahil9001** [comment]
I've did the same thing you suggested but its still not working @codesankalp

File: .github/workflows/tests.yml
Comment (codesankalp) [inline .github/workflows/tests.yml]:
```suggestion
      run: echo -e "SECRET_KEY=${{ secrets.GITHUB_TOKEN }}" >> .env
```
Try with this @sahil9001.

**Review by codesankalp — DISMISSED** [review]
Approving this PR. :+1: @sahil9001  
@isabelcosta Needs your review and what's your opinion in adding test in this or not?

File: .github/workflows/tests.yml
Comment (isabelcosta) [inline .github/workflows/tests.yml]:
```suggestion
      run: echo -e "SECRET_KEY=${{ secrets.SECRET_KEY }}" >> .env
```
Since this can be a random string 🤔 
Github token may or may not be needed in the future because of GitHub logins, not sure 🤔 So it's better to name this on repositories end, in a different way.

**Comment by isabelcosta** [comment]
I think tests, in this case, is not straight forward, since it's a condition on settings.py and not a function we can unit test 🤔 
I will approve and merge, since you @codesankalp already tested it

**Review by isabelcosta — APPROVED** [review]
Good work @sahil9001 :) Thank you for contributing!

**Pull request merged**
